src/app/medicine/medicine.cpp

```diff
@@ -1,15 +1,39 @@
 #include "medicine.hpp"
+#include <ctime>
+#include <iostream>
 
-Medicine::Medicine(const std::string& name, const double price, const int amount, const int expiration_date,
+Medicine::Medicine(const std::string& name, const double price, const int amount,
+                   const std::chrono::year_month_day expiration_date,
                    const std::pair<double, double>& storage_temperature, const std::vector<std::string>& composition)
     : Product(name, price, amount), expiration_date_{expiration_date}, storage_temperature_{storage_temperature},
       composition_{composition}
 {
 }
-bool Medicine::isExpired()
+bool Medicine::isExpired() const
 {
-    // TODO: Write this function when Date will be ready
-    return false;
+    std::time_t t = std::time(0);
+    std::tm* now = std::localtime(&t);
+    int currentYear = now->tm_year + 1900;
+    unsigned currentMonth = now->tm_mon + 1;
+    unsigned currentDay = now->tm_mday;
+
+    if (static_cast<int>(expiration_date_.year()) < currentYear)
+    {
+        return true;
+    }
+    else if (static_cast<int>(expiration_date_.year()) >= currentYear and
+             static_cast<unsigned>(expiration_date_.month()) < currentMonth)
+    {
+        return true;
+    }
+    else if (static_cast<int>(expiration_date_.year()) >= currentYear and
+             static_cast<unsigned>(expiration_date_.month()) >= currentMonth and
+             static_cast<unsigned>(expiration_date_.day()) < currentDay)
+    {
+        return true;
+    }
+    else
+        return false;
```

Comment:
Isn't chrono a better choice here? 
I assume [timepoint](https://en.cppreference.com/w/cpp/chrono/time_point) and [duration](https://en.cppreference.com/w/cpp/chrono/duration) would help.

Especially with duration we would only need only expire data and comparison if duration between today and expire date is <=0

Reply:
Just a thought I am not sure it ould work. Maybe create currrent day as year_mont_day and check if its >= expiration_date?

Reply:
Done